etl-pipeline/processes/ingest/gutenberg.py

```diff
@@ -19,87 +23,73 @@ class GutenbergProcess():
+            start_timestamp=utils.get_start_datetime(
+                process_type=self.params.process_type,
+                ingest_period=self.params.ingest_period,
+            ),
             offset=self.params.offset,
             limit=self.params.limit,
         )
 
+        count = 0
         for record_mapping in records:
-            self.store_epubs(record_mapping)
-
+            self.add_epub_links(record_mapping.record)
             try:
                 self.add_cover(record_mapping)
             except Exception:
                 logger.warning(f'Unable to store cover for {record_mapping.record.source_id}')
 
-            self.record_buffer.add(record_mapping.record)
-
-        self.record_buffer.flush()
-
-        logger.info(f'Ingested {self.record_buffer.ingest_count} Gutenberg records')
-
-        return self.record_buffer.ingest_count
-
-    def store_epubs(self, gutenberg_record: GutenbergMapping):
-        epub_parts = []
-
-        for part in gutenberg_record.record.parts:
-            epub_id_parts = re.search(r'\/([0-9]+).epub.([a-z]+)$', part.url)
-            gutenberg_id = epub_id_parts.group(1)
-            gutenberg_type = epub_id_parts.group(2)
-
-            if json.loads(part.flags).get('download', False) is True:
-                epub_path = f'epubs/{part.source}/{gutenberg_id}_{gutenberg_type}.epub'
-                epub_url = get_stored_file_url(self.file_bucket, epub_path)
-
-                epub_parts.append(str(Part(
-                    index=part.index,
-                    source=part.source,
-                    url=epub_url,
-                    file_type=part.file_type,
-                    flags=part.flags
-                )))
-
-                self.rabbitmq_manager.send_message_to_queue(self.file_queue, self.file_route, get_file_message(part.url, epub_path))
-            else:
-                container_path = f'epubs/{part.source}/{gutenberg_id}_{gutenberg_type}/META-INF/container.xml'
-                manifest_path = f'epubs/{part.source}/{gutenberg_id}_{gutenberg_type}/manifest.json'
-
-                epub_parts.append(str(Part(
-                    index=part.index,
-                    source=part.source,
-                    url=get_stored_file_url(self.file_bucket, container_path),
-                    file_type='application/epub+xml',
-                    flags=part.flags
-                )))
-
-                epub_parts.append(str(Part(
-                    index=part.index,
-                    source=part.source,
-                    url=get_stored_file_url(self.file_bucket, manifest_path),
-                    file_type='application/epub+xml',
-                    flags=part.flags
-                )))
-
-        gutenberg_record.record.has_part = epub_parts
+            count += 1
+            yield record_mapping.record
+
+        return count
+
+    def add_epub_links(self, record: Record):
```

Comment:
We could consider moving this to the mapper

Reply:
done!